spec/system/oauth_spec.rb

```diff
@@ -42,6 +45,79 @@
       # It does not grant the app access to the account
       expect(Doorkeeper::AccessGrant.exists?(application: client_app, resource_owner_id: user.id)).to be false
     end
+
+    # The tests in this context ensures that requests without PKCE parameters
+    # still work; In the future we likely want to force usage of PKCE for
+    # security reasons, as per:
+    #
+    # https://www.ietf.org/archive/id/draft-ietf-oauth-security-topics-27.html#section-2.1.1-9
+    context 'when not using PKCE' do
+      it 'does not include the PKCE values in the hidden inputs' do
+        subject
+
+        code_challenge_inputs = all('.oauth-prompt input[name=code_challenge]', visible: false)
```

Comment:
Small style feedback -- pull this and the line below out into private methods instead of local vars (the same selector is used in this example and the one in the context below)?

Reply:
I get and respect your point. I am however not convinced that this would help the readability of this spec. I like using the syntax possibilities rspec gives me (sometimes a bit too much :grimacing:), but mostly believe in "stupid" and explicit test code, even if that is not super "DRY".